### PR TITLE
Add SIG Architecture Code Organization spotlight article

### PR DESCRIPTION
**Pull request by fsmunoz**
SIG Architecture: Code Organization spotlight article.

Closes: #482

### CODE REVIEW

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+Furthermore, some of these changes are also candidates for
+[backporting](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md)
+to our supported release branches.
+
+**FSM**: Is there any major project or theme the subproject is working on right now that you would
+like to highlight?
+
+**MJ**: I think one very interesting and immensely useful change that has been recently added (and I
+take the opportunity to specifically highlight the work of [Tim Hochin](https://github.com/thockin)
```
Comment (nimbinatus) [inline content/en/blog/2024/sig-arch-code-spotlight.md:118]:
```suggestion
take the opportunity to specifically highlight the work of [Tim Hockin](https://github.com/thockin)
```

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+repo](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/4402-go-workspaces). A
+lot of our current tooling for dependency management and code publishing, as well as the experience
+of editing code in the Kubernetes repo, can be significantly improved by this change.
+
+## Wrapping up
+
+**FSM**: How would someone interested in the topic start helping the subproject?
+
+**MJ**: The first step, as is the first step with any project in Kubernetes, is to join our slack:
+slack.k8s.io, and after that join the `#k8s-code-organization` channel. There is also a
```
Comment (nimbinatus) [inline content/en/blog/2024/sig-arch-code-spotlight.md:129]: Any reason not to make the Slack link an actual link?
```suggestion
[slack.k8s.io](https://slack.k8s.io), and after that join the `#k8s-code-organization` channel. There is also a
```

**Review by nimbinatus — CHANGES_REQUESTED** [review]
Not going through a detailed grammar review after last time :) Left a couple questions and fixed a spelling error. All links have been checked.

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+[slack.k8s.io](https://slack.k8s.io), and after that join the `#k8s-code-organization` channel. There is also a
+[code-organization office
+hours](https://github.com/kubernetes/community/tree/master/sig-architecture#meetings) that takes
+place that you can choose to attend. Timezones are hard, so feel free to also look at the recordings
+or meeting notes and follow up on slack!
+
+**FSM**: Excellent, thank you! Any final comments you would like to share?
+
+**MJ**: The Code Organization subproject always needs help! Especially areas like the publishing
+bot, so don’t hesitate to get involved in the `#k8s-code-organization` slack channel.
```
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:138]: (nit)
```suggestion
bot, so don’t hesitate to get involved in the `#k8s-code-organization` Slack channel.
```

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+[backporting](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md)
+to our supported release branches.
+
+**FSM**: Is there any major project or theme the subproject is working on right now that you would
+like to highlight?
+
+**MJ**: I think one very interesting and immensely useful change that has been recently added (and I
+take the opportunity to specifically highlight the work of [Tim Hockin](https://github.com/thockin)
+on this) is the introduction of [Go workspaces to the Kubernetes
+repo](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/4402-go-workspaces). A
```
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:120]: Could instead link to [`/blog/2024/03/19/go-workspaces-in-kubernetes/`](https://k8s.dev/blog/2024/03/19/go-workspaces-in-kubernetes/), the article about this.

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+that is needed to help Kubernetes [stay on supported versions of
+Go](https://github.com/kubernetes/enhancements/tree/cf6ee34e37f00d838872d368ec66d7a0b40ee4e6/keps/sig-release/3744-stay-on-supported-go-versions),
+and it is through this that I am involved with the Code Organization subproject of SIG Architecture.
+
+**FSM**: A project the size of Kubernetes must have unique challenges in terms of code organization
+-- is this a fair assumption?  If so, what would you pick as some of the main challenges that are
+specific to Kubernetes?
+
+**MJ**: That’s a fair assumption! The first interesting challenge comes from the sheer size of the
+Kubernetes codebase. We have ~2.2 million lines of Go code (which is steadily decreasing thanks to
```
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:34]: We an use the actual math symbol:
```suggestion
Kubernetes codebase. We have ≅2.2 million lines of Go code (which is steadily decreasing thanks to
```

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+slug: sig-architecture-code-spotlight-2024
+date: 2024-04-04
+author: "Frederico Muñoz (SAS Institute)"
+---
+
+_This is the third interview of a SIG Architecture Spotlight series that will cover the different
+subprojects. We will cover the [SIG Architecture: Code
+Organization](https://github.com/kubernetes/community/blob/master/sig-architecture/README.md#code-organization-1)_
+
+In this SIG Architecture spotlight we talked with [Madhav
```
Comment (sftim) [inline content/en/blog/2024/sig-arch-code-spotlight.md:13]: (nit)
```suggestion
In this SIG Architecture spotlight I talked with [Madhav
```
There were two people talking, one was Madhav, so you'd use first person singular.

**Review by sftim — COMMENTED** [review]
LGTM apart from publication date.

Very small nits.

/hold
for assignment of publication date.

How about Thursday (2024-04-11)?

File: content/en/blog/2024/sig-arch-code-spotlight.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+layout: blog
+title: "Spotlight on SIG Architecture: Code Organization"
+slug: sig-architecture-code-spotlight-2024
+date: 2024-04-04
+author: "Frederico Muñoz (SAS Institute)"
+---
+
+_This is the third interview of a SIG Architecture Spotlight series that will cover the different
+subprojects. We will cover the [SIG Architecture: Code
+Organization](https://github.com/kubernetes/community/blob/master/sig-architecture/README.md#code-organization-1)_
```
[the remark this answers is not in the document]
Comment (fsmunoz) [reply content/en/blog/2024/sig-arch-code-spotlight.md:11]: Done.

File: content/en/blog/2024/sig-arch-code-spotlight.md (hunk 6 above)
Comment (fsmunoz) [reply content/en/blog/2024/sig-arch-code-spotlight.md:13]: Yes, old habits that are not adequate here.

File: content/en/blog/2024/sig-arch-code-spotlight.md (hunk 4 above)
Comment (fsmunoz) [reply content/en/blog/2024/sig-arch-code-spotlight.md:120]: Yes much better. Changed in last commit.

**Comment by fsmunoz** [comment]
@nimbinatus @sftim I think this is good to go, I've reflected all suggestions, updated the publication date for the suggested one, and squashed it in case it really is the final version.

**Comment by sftim** [comment]
@dims want to send in a PR to do a fixup here too?

(wrt https://github.com/kubernetes/website/pull/45853)

**Comment by dims** [comment]
may i defer to @fsmunoz please?

**Comment by fsmunoz** [comment]
@dims thanks for the fix, I can add it here if it helps, sure.

**Pull request merged**